beacon_node/eth1/src/http.rs

```diff
@@ -91,14 +92,29 @@ pub async fn get_network_id(endpoint: &str, timeout: Duration) -> Result<Eth1Id,
 
 /// Get the eth1 chain id of the given endpoint.
 pub async fn get_chain_id(endpoint: &str, timeout: Duration) -> Result<Eth1Id, String> {
-    let response_body = send_rpc_request(endpoint, "eth_chainId", json!([]), timeout).await?;
-    hex_to_u64_be(
-        response_result(&response_body)?
-            .ok_or("No result was returned for chain id")?
-            .as_str()
-            .ok_or("Data was not string")?,
-    )
-    .map(Into::into)
+    let response_body: String =
+        send_rpc_request(endpoint, "eth_chainId", json!([]), timeout).await?;
+
+    /* extract response text here */
+    let response_result: Result<Result<Value, Value>, String> =
+        response_result_or_error(&response_body);
```

Comment:
Can directly return the result here so you don't have to handle the outer error case separately
```suggestion
        response_result_or_error(&response_body)?;
```

Reply:
Thanks again for this! I should've checked the return types more thoroughly. Consult 867afdb.